packages/cache/src/internal/cacheUtils.ts

```diff
@@ -0,0 +1,99 @@
+import * as core from '@actions/core'
+import * as exec from '@actions/exec'
+import * as glob from '@actions/glob'
+import * as io from '@actions/io'
+import * as fs from 'fs'
+import * as path from 'path'
+import * as util from 'util'
+import {v4 as uuidV4} from 'uuid'
+import {CacheFilename, CompressionMethod} from './constants'
+
+// From https://github.com/actions/toolkit/blob/master/packages/tool-cache/src/tool-cache.ts#L23
+export async function createTempDirectory(): Promise<string> {
+  const IS_WINDOWS = process.platform === 'win32'
+
+  let tempDirectory: string = process.env['RUNNER_TEMP'] || ''
+
+  if (!tempDirectory) {
```

Comment:
should we create a `@actions/core` function for this, we use it in other places? We can create an issue for it and solve it sometime in the future if you want :) https://github.com/actions/toolkit/blob/57d20b4db494c25af8d2f3d9323650044610e531/packages/tool-cache/src/tool-cache.ts#L555

Is there a reason we expect `RUNNER_TEMP` to not be defined? IIRC, it should always be defined, it may be okay to error if it isn't.

Reply:
It should always be defined see Josh's comment above but I think we just want to make sure it doesn't fail cache if the folder is for some reason deleted (maybe in self-hosted runner).